### PR TITLE
feat: added support of passing labels to pre translate command

### LINKED ISSUE #640: Add the '--label' parameter to the 'pre-translate' command
**Description**

Crowdin has recently enhanced the [Apply Pre-Translation](https://developer.crowdin.com/api/v2/#operation/api.projects.pre-translations.post) API to allow passing labels to filter strings for pre-translation.

The new `--label` option should be added to the `crowdin pre-translate` command as well. 

```yml
SYNOPSIS:
crowdin pre-translate [CONFIG OPTIONS] [OPTIONS]

DESCRIPTION:
Pre-translate files via Machine Translation (MT) or Translation Memory (TM)

OPTIONS:
...
      --label=...         Label identifier. Could be specified multiple times
...
```

It should accept **string** values of label names. Use should be able to specify multiple labels simultaneously (e.g. `crowdin pre-translate --label android --label ios --label web ...`)

**Additional Context**

Similar commands for reference: `BundleAddSubcommand`, `StringAddSubcommand`, `TaskAddSubcommand`.
Documentation:[`crowdin pre-translate`](https://crowdin.github.io/crowdin-cli/commands/crowdin-pre-translate)
Blocked by: https://github.com/crowdin/crowdin-api-client-java/issues/188

### PR DESCRIPTION
**Pull request by debanjanc01**
This closes https://github.com/crowdin/crowdin-cli/issues/640

### CODE REVIEW

File: src/main/java/com/crowdin/cli/commands/picocli/PreTranslateSubcommand.java [hunk 1]
```diff
@@ -49,6 +49,9 @@ public class PreTranslateSubcommand extends ActCommandWithFiles {
     @CommandLine.Option(names = {"--plain"}, descriptionKey = "crowdin.list.usage.plain")
     protected boolean plainView;
 
+    @CommandLine.Option(names = {"--label"}, descriptionKey = "params.label", paramLabel = "...", order = -2)
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/commands/picocli/PreTranslateSubcommand.java]: This is the wrong description for this parameter within this command. You need to create a new message

```properties
crowdin.pre-translate.label=Label identifier. Could be specified multiple times
```

File: src/main/java/com/crowdin/cli/commands/actions/PreTranslateAction.java [hunk 2]
```diff
@@ -141,6 +145,21 @@ private List<Long> prepareFileIds(Outputter out, PropertiesWithFiles pb, Crowdin
...
+            labelNames.stream()
+                    .distinct()
+                    .forEach(labelName -> labels.computeIfAbsent(labelName, (title) -> client.addLabel(RequestBuilder.addLabel(title)).getId()));
+            return labelNames.stream()
+                    .map(labels::get)
+                    .collect(Collectors.toList());
+        } else {
+            return null;
+        }
+    }
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/cli/commands/actions/PreTranslateAction.java]: I just checked this feature in more detail and realized that this functionality should behave a bit differently than it does in the string add command. It shouldn't add a new label if one is missing in the Crowdin project. This command should give a warning if the non-existent label is passed.

Comment (debanjanc01) [reply src/main/java/com/crowdin/cli/commands/actions/PreTranslateAction.java]: I've added this in the latest commit.

File: src/main/java/com/crowdin/cli/commands/picocli/PreTranslateSubcommand.java (hunk 1 above)
Comment (debanjanc01) [reply src/main/java/com/crowdin/cli/commands/picocli/PreTranslateSubcommand.java]: That makes sense. Added this description in the latest message.

File: src/main/resources/messages/messages.properties [hunk 3]
```diff
@@ -671,6 +672,7 @@ message.comment.added=@|green Comment|@ @|yellow #%s|@ @|green,bold '%s'|@ @|gre
 
 message.pre_translate.local_files_message=Out of %d files, only %d of them were found locally. Use '--verbose' to list them
 message.pre_translate.local_files_message_verbose=Out of %d files, %d of them were found locally:
+message.pre_translate.missing_label=Label '%s' passed is invalid. Skipping it
```
Comment (andrii-bodnar) [inline src/main/resources/messages/messages.properties]:
```suggestion
message.pre_translate.missing_label=The '%s' label is missing in the Crowdin project
```

**Comment by debanjanc01** [comment]
Hi @andrii-bodnar shall I squash all the commits into a single one?

**Comment by andrii-bodnar** [comment]
No, it's not needed, I will squash them during the merge

**Comment by debanjanc01** [comment]
@andrii-bodnar alright, thanks!

**Review by andrii-bodnar — APPROVED** [review]
@debanjanc01 thanks a lot for the contribution! 🚀

**Pull request merged**